transformer/classes/dockerfile/pythondockerfilegenerator.go

```diff
@@ -0,0 +1,213 @@
+	}
+	return nil, nil
+}
+
+// Transform transforms the artifacts
+func (t *PythonDockerfileGenerator) Transform(newArtifacts []transformertypes.Artifact, oldArtifacts []transformertypes.Artifact) ([]transformertypes.PathMapping, []transformertypes.Artifact, error) {
+	pathMappings := []transformertypes.PathMapping{}
+	artifactsCreated := []transformertypes.Artifact{}
+	for _, a := range newArtifacts {
+		relSrcPath, err := filepath.Rel(t.Env.GetEnvironmentSource(), a.Paths[artifacts.ProjectPathPathType][0])
+		if err != nil {
+			logrus.Errorf("Unable to convert source path %s to be relative : %s", a.Paths[artifacts.ProjectPathPathType][0], err)
+		}
+		var sConfig artifacts.ServiceConfig
+		err = a.GetConfig(artifacts.ServiceConfigType, &sConfig)
+		if err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", sConfig, err)
+			continue
+		}
+		sImageName := artifacts.ImageName{}
+		err = a.GetConfig(artifacts.ImageNameConfigType, &sImageName)
+		if err != nil {
+			logrus.Debugf("unable to load config for Transformer into %T : %s", sImageName, err)
+		}
+		var pythonConfig PythonTemplateConfig
+		pythonConfig.AppName = a.Name
+		if a.Name == djangoService {
+			pythonConfig.IsDjangoProject = true
+		}
+		mainPythonFileRelPath, err := filepath.Rel(a.Paths[artifacts.ProjectPathPathType][0], a.Paths[MainPythonFilePathType][0])
+		if err != nil {
+			logrus.Errorf("Error in getting the relative path %s", err)
+		} else {
+			pythonConfig.MainScriptRelPath = mainPythonFileRelPath
+		}
+		var detectedPorts []int32
+		detectedPorts = append(detectedPorts, 8080) //TODO: Write parser to parse and identify port
+		pythonConfig.Port = commonqa.GetPortForService(detectedPorts, a.Name)
+		requirementsTxtFiles, err := common.GetFilesByName(a.Paths[artifacts.ProjectPathPathType][0], []string{requirementsTxtFile}, nil)
+		if err != nil {
+			logrus.Debugf("Cannot get the requirements.txt file: %s", err)
+		}
+		if len(requirementsTxtFiles) == 1 {
+			requirementsTxt, err := filepath.Rel(a.Paths[artifacts.ProjectPathPathType][0], requirementsTxtFiles[0])
+			if err != nil {
+				logrus.Errorf("Error in getting the relative path %s", err)
+			} else {
+				pythonConfig.RequirementsTxt = requirementsTxt
+			}
+		}
+		if sImageName.ImageName == "" {
+			sImageName.ImageName = common.MakeStringContainerImageNameCompliant(sConfig.ServiceName)
+		}
+		pathMappings = append(pathMappings, transformertypes.PathMapping{
+			Type:     transformertypes.SourcePathMappingType,
+			DestPath: common.DefaultSourceDir,
+		}, transformertypes.PathMapping{
+			Type:           transformertypes.TemplatePathMappingType,
+			SrcPath:        filepath.Join(t.Env.Context, t.Config.Spec.TemplatesDir),
+			DestPath:       filepath.Join(common.DefaultSourceDir, relSrcPath),
+			TemplateConfig: pythonConfig,
+		})
+		paths := a.Paths
+		paths[artifacts.DockerfilePathType] = []string{filepath.Join(common.DefaultSourceDir, relSrcPath, "Dockerfile")}
```

Comment:
Do we want to use the const for Dockerfilename

Reply:
Yes sure :). Done.